From 6521d31ad06796bdd5e99b4f4c3acbbe92db2aa6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim.muller@collabora.co.uk>
Date: Mon, 3 Jan 2011 01:18:06 +0000
Subject: [PATCH 30/44] v4l2src: undo presumably accidental enablement of the GstXOverlay interface

Looks like this got enabled by accident when adding it to v4l2sink,
so undo this for now. Not sure it makes much sense in a GStreamer
context with current hardware.
---
 sys/v4l2/gstv4l2src.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index 0e9ced9..0a468c2 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -44,6 +44,8 @@
 #include <config.h>
 #endif
 
+#undef HAVE_XVIDEO
+
 #include <string.h>
 #include <sys/time.h>
 #include "v4l2src_calls.h"
@@ -143,6 +145,7 @@ gst_v4l2src_init_interfaces (GType type)
     NULL,
   };
 #ifdef HAVE_XVIDEO
+  /* FIXME: does GstXOverlay for v4l2src make sense in a GStreamer context? */
   static const GInterfaceInfo v4l2_xoverlay_info = {
     (GInterfaceInitFunc) gst_v4l2src_xoverlay_interface_init,
     NULL,
-- 
1.7.1

